app/scripts/modules/core/src/managed/UnmanagedMessage.tsx

```diff
@@ -0,0 +1,37 @@
+import { ApolloError } from '@apollo/client';
+import React from 'react';
+
+import { SETTINGS } from 'core/config';
+
+const defaultGettingStartedUrl = 'https://www.spinnaker.io/guides/user/managed-delivery/getting-started/';
+
+export const UnmanagedMessage = () => {
+  const gettingStartedLink = SETTINGS.managedDelivery?.gettingStartedUrl || defaultGettingStartedUrl;
+  return (
+    <div style={{ width: '100%' }}>
+      Welcome! This application does not have any environments or artifacts. Check out the{' '}
+      <a href={gettingStartedLink} target="_blank">
+        getting started guide
+      </a>{' '}
+      to set some up!
+    </div>
+  );
+};
+
+interface IQueryErrorHandlerProps {
+  hasApplicationData: boolean;
+  error: ApolloError;
+}
+
+export const QueryErrorHandler = ({ hasApplicationData, error }: IQueryErrorHandlerProps) => {
+  if (!hasApplicationData) {
+    return <UnmanagedMessage />;
+  }
+  console.warn(error);
+  return (
+    <div style={{ width: '100%' }}>
+      Failed to load environments data, please refresh and try again.
+      <p>{error.message}</p>
+    </div>
+  );
+};
```

Comment:
I think this component should be in a separate file since its a unique error state that is unrelated to being unmanaged (or it seems like its unrelated to being unmanaged from my context :) )